adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -436,6 +436,8 @@ protected void onRestoreInstanceState(final Bundle savedInstanceState) {
     private void setupWebView() {
         final WebSettings webSettings = mWebView.getSettings();
         webSettings.setJavaScriptEnabled(true);
+        // Explicitly set false, may be true for older devices (ICS MR1 & earlier)
+        webSettings.setAllowUniversalAccessFromFileURLs(false);
```

Comment:
Are there other WebSettings that we should manipulate?

There's allowFileAccess, allowFileAccessFromFileUrls, etc. which have similar statements about their usability and functionality on different versions, but I don't see a statement about what the defaults are... 

https://developer.android.com/reference/android/webkit/WebSettings

Also, is it possible to test this?

Reply:
Let me dig deeper into that and see what I can find

Reply:
| API                                     | Default Behavior                                                                                                                        | Enabled in ADAL | Desired state in ADAL |
|-----------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|-----------------|-----------------------|
| [`setAllowContentAccess()`](https://developer.android.com/reference/android/webkit/WebSettings#setAllowContentAccess(boolean))               | Enabled                                                                                                                                 | Disabled        | Disabled              |
| [`setAllowFileAccess()`](https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccess(boolean))                  | API Level Dependent <br/> Enabled by default for API <=29 <br/> Disabled by default for API 30+                                         | Enabled         | Disabled              |
| [`setAllowFileAccessFromFileURLs()`](https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccessFromFileURLs(boolean))      | Ignored if `getAllowUniversalAccessFromFileURLs()` is `true` <br/> Enabled by default for API <=15 <br/> Disabled by default for API 16+ | Disabled        | Disabled              |
| [`setAllowUniversalAccessFromFileURLs()`](https://developer.android.com/reference/android/webkit/WebSettings#setAllowUniversalAccessFromFileURLs(boolean)) | Enabled by default for API <=15 <br/> Disabled by default for API 16+                                                                   | Disabled        | Disabled              |

Reply:
@AdamBJohnsonx See above table, since we're now targeting API 27 in our build, we can actually no-op all of these _except_ `setAllowFileAccess()` which is still enabled by default.

From the docs:
```
To prevent possible security issues targeting Build.VERSION_CODES.Q and earlier, you should explicitly set this value to false.
```

So I am going to repurpose this PR to instead disable file access, and will close the _original_ issue as it no longer applies.

Reply:
@AdamBJohnsonx ADAL `minSdk` is currently 16, with `targetSdk` at 27. So these are noops for everything _except_ `setAllowFileAccess()`.

From the docs:
> To prevent possible security issues targeting Build.VERSION_CODES.Q and earlier, you should explicitly set this value to false.

So I am going to close the original issue, and repurpose this PR to address `setAllowFileAccess()`

Reply:
(I am not thinking I will implement a specific test for this, given it's a built-in)

Reply:
See 02f83275bc60159ceb97fb72ada5a60b66bac529